File: js-pr/56/javascript/code-quality-performance/collection_processor_06.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function processItems(items) {
    let itemArray = Array.from(items);
    for (let i = 0; i < itemArray.length; i++) {
        console.log(itemArray[i]);
    }
}
// {/fact}

File: java-pr/49/java/performance/CollectionProcessor06.java
// {fact rule=performance@v1.0 defects=1}
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class CollectionProcessor06 {
    private Database database = new Database();
    
    public void processItems(Collection<String> items) {
        String[] array = items.toArray(new String[0]);
        for (int i = 0; i < array.length; i++) {
            System.out.println(array[i]);
        }
    }
    
    private static class Database {
        public User findUserById(int id) { return new User(); }
        public List<User> findAllUsers() { return new ArrayList<>(); }
    }
    
    private static class User {}
}
// {/fact}